pywbem_mock/_mainprovider.py

```diff
@@ -811,152 +790,247 @@ def remove_namespace(self, namespace):
+              :term:`DSP0200` (server-implemented default is `False`).
+
+            Note, the semantics of the `DeepInheritance` parameter in
+            :meth:`~pywbem.WBEMConnection.EnumerateInstances` is different.
+
         Returns:
 
-            return tuple including list of classnames
+            A list of :term:`unicode string` objects that are the class names
+            of the enumerated classes
 
         Raises:
 
             CIMError: CIM_ERR_INVALID_NAMESPACE: invalid namespace,
             CIMError: CIM_ERR_INVALID_CLASS: class defined by the classname
             parameter does not exist
         """
-        namespace = self.namespace
 
         # Validate namespace and get class_store for this namespace
+        self.validate_namespace(namespace)
         class_store = self.get_class_store(namespace)
 
-        classname = params.get('ClassName', None)
-        if classname:
-            assert(isinstance(classname, CIMClassName))
-            if not class_store.exists(classname.classname):
+        if ClassName:
+            assert(isinstance(ClassName, six.string_types))
+            if not class_store.exists(ClassName):
                 raise CIMError(
                     CIM_ERR_INVALID_CLASS,
                     _format("The class {0!A} defined by 'ClassName' parameter "
                             "does not exist in namespace {1!A}",
-                            classname, namespace))
-
-        clns = self._get_subclass_names(classname, class_store,
-                                        params['DeepInheritance'])
-        rtn_clns = [
-            CIMClassName(cn, namespace=namespace, host=self.host)
-            for cn in clns]
+                            ClassName, namespace))
 
-        return self._make_tuple(rtn_clns)
+        # Return list of subclass names
+        return self._get_subclass_names(ClassName, class_store, DeepInheritance)
 
     ####################################################################
     #
     #   GetClass server request
     #
     ####################################################################
 
-    def GetClass(self, **params):
-        # pylint: disable=invalid-name
+    def GetClass(self, namespace, ClassName, LocalOnly=None,
+                 IncludeQualifiers=None, IncludeClassOrigin=None,
+                 PropertyList=None):
         """
-        Implements a mock server responder for
+        Implements a  WBEM provider responder  for
         :meth:`~pywbem.WBEMConnection.GetClass
 
         Retrieve a CIM class from the local repository.
 
-        For a description of the parameters, see
-        :meth:`pywbem.WBEMConnection.GetClass`.
+        Parameters:
+
+          namespace (:term:`string`):
+            Name of the namespace from which the class will be retrieved
+            (case independent).
+
+            Leading and trailing slash characters will be stripped. The lexical
+            case will be preserved.
```

Comment:
The namespace is used only for lookup, it is not stored. The language used to explain the handling of leading and trailing slashes and the lexical case are good when it is stored, but confusing when it is used only for lookup. Better talk about ignoring slashes and case.

Reply:
Dups previous comments.  All description of namespace parameter was modified.